apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -520,16 +520,27 @@ defmodule Champions.Battle.Simulator do
   # Check if the unit can cast their basic skill this step.
   defp can_cast_basic_skill(unit), do: unit.basic_skill.remaining_cooldown <= 0
 
+  defp choose_targets(caster, targeting_strategy, state) do
+    targeteable_units =
+      state.units
+      |> Enum.filter(fn {_, unit} -> not Enum.any?(unit.tags, &(&1.tag == "Untargetable")) end)
+
+    state_with_targeteable_units =
+      Map.put(state, :units, targeteable_units)
+
+    choose_targets_by_strategy(caster, targeting_strategy, state_with_targeteable_units)
```

Comment:
Is it really necessary to include the targetable units inside the state? maybe it could be a new parameter of the `choose_targets_by_strategy`, just wondering